### PR TITLE
 Updated the api_gateway.tf and alb.tf

### PR DESCRIPTION
**Pull request by abibat-adesina**

### CODE REVIEW

File: alb.tf [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 
-# Use the default ALB that is pre-provisioned as part of the account creation
-# This ALB has all traffic on *.LICENSE-PLATE-ENV.nimbus.cloud.gob.bc.ca routed to it
-data "aws_alb" "main" {
-  name = var.alb_name
-}
+  name               = var.app_name
+  internal           = true
+  subnets            = module.network.aws_subnets.web.ids
+  security_groups    = module.network_load_balancer_arns
```
Comment (wrnu) [inline alb.tf]:
```suggestion
  security_groups    = module.network.aws_security_groups.web.ids
```

File: alb.tf [hunk 2]
```diff
@@ -1,17 +1,25 @@
...
-  port              = 443
 }
+resource "aws_lb_listener" "internal" {
+  load_balancer_arn = aws_lb.app-alb.arn
+  port              = "80"
+  protocol          = "TCP"
 
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.example.arn
```
Comment (wrnu) [inline alb.tf]:
```suggestion
    target_group_arn = aws_lb_target_group.app.arn
```

File: alb.tf [hunk 3]
```diff
@@ -36,14 +44,9 @@ resource "aws_alb_target_group" "app" {
...
     type             = "forward"
     target_group_arn = aws_alb_target_group.app.arn
   }
 
-  condition {
-    host_header {
-      values = [for sn in var.service_names : "${sn}.*"]
-    }
-  }
 }
```
Comment (wrnu) [inline alb.tf]: We can remove the entire `resource "aws_lb_listener_rule" "host_based_weighted_routing"` it is covered by the default action. Sorry my bad

**Review by wrnu — CHANGES_REQUESTED** [review]

File: alb.tf (hunk 3 above)
Comment (wrnu) [reply alb.tf]: to be clear it is handled by the default action on the listener

**Comment by abibat-adesina** [comment]
Updated to include the corrections

File: alb.tf [hunk 4]
```diff
@@ -41,12 +41,9 @@ resource "aws_alb_target_group" "app" {
   tags = local.common_tags
 }
 
-resource "aws_lb_listener_rule" "host_based_weighted_routing" {
-  listener_arn = data.aws_alb_listener.front_end.arn
-
   default_action {
     type             = "forward"
     target_group_arn = aws_alb_target_group.app.arn
   }
```
Comment (wrnu) [inline alb.tf]:
```suggestion
```

This should also be removed

**Review by wrnu — CHANGES_REQUESTED** [review]
Please make sure to run `terraform fmt .` in the directory containing the terraform files before you commit. This will ensure that all files are formatted similarly.

File: alb.tf [hunk 5]
```diff
@@ -33,17 +41,9 @@ resource "aws_alb_target_group" "app" {
...
 }
 
-resource "aws_lb_listener_rule" "host_based_weighted_routing" {
-  listener_arn = data.aws_alb_listener.front_end.arn
-
-  action {
+  default_action {
     type             = "forward"
     target_group_arn = aws_alb_target_group.app.arn
   }
```
Comment (wrnu) [inline alb.tf]:
```suggestion
```

This should be removed

**Review by wrnu — CHANGES_REQUESTED** [review]

**Comment by abibat-adesina** [comment]
Terraform fmt ran successfully and default code removed

**Review by wrnu — APPROVED** [review]

**Pull request merged**